src/config/format.rs

```diff
@@ -0,0 +1,261 @@
+//! Support for loading configs from multiple formats.
+
+#![deny(missing_docs, missing_debug_implementations)]
+
+use serde::de;
+use std::path::Path;
+
+/// The format used to represent the configuration data.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum Format {
+    /// The format could not be determined.
+    Unknown,
+    /// TOML format is used.
+    TOML,
+    /// JSON format is used.
+    JSON,
+    /// YAML format is used.
+    YAML,
+}
+
+impl Default for Format {
+    fn default() -> Self {
+        Format::TOML
+    }
+}
+
+impl Format {
+    /// Returns the format as is unless it's unknown.
+    /// If the format is unknown - executes the function and return the result
+    /// of the function.
+    pub fn known_or<F>(self, f: F) -> Self
+    where
+        F: FnOnce() -> Self,
+    {
+        match self {
+            Format::Unknown => f(),
+            _ => self,
+        }
+    }
+}
+
+/// Obtain the format from the file path using extension as a hint.
+pub fn from_path(path: impl AsRef<Path>) -> Format {
```

Comment:
Why not `impl From<AsRef<Path>> for Format`?

Reply:
Indeed, fixed!